### PR TITLE
Support string literals

### LINKED ISSUE #72: Add support for string literals
This generates an error:
```
const S: &str = "Hello";
```

Error:
```
Unsupported constant: "ConstValue::Slice: Slice { data: ConstAllocation { .. }, start: 0, end: 98 }"
```

### PR DESCRIPTION
**Pull request by Nadrieril**
Thanks to https://github.com/hacspec/hax/pull/649, hax supports string literals as they show up in MIR.

Fixes #72

### CODE REVIEW

File: charon-ml/src/NameMatcher.ml [hunk 1]
```diff
@@ -63,6 +63,8 @@ let literal_to_string (c : print_config) (l : literal) : string =
           raise (Failure "TODO")
       | TkPretty -> "'" ^ String.make 1 x ^ "'"
       | TkName -> String.make 1 x)
+  | LString s -> "\"" ^ s ^ "\""
+  | LByteStr bs -> "[" ^ (String.concat ", " (List.map string_of_int bs)) ^ "]"
```
Comment (Nadrieril) [inline charon-ml/src/NameMatcher.ml:67]: Is this corret? What is this printing used for?

File: charon-ml/src/PrintValues.ml [hunk 2]
```diff
@@ -33,3 +33,5 @@ let literal_to_string (lit : literal) : string =
   | VScalar sv -> scalar_value_to_string sv
   | VBool b -> Bool.to_string b
   | VChar c -> String.make 1 c
+  | VString s -> "\"" ^ s ^ "\""
+  | VByteStr bs -> "[" ^ (String.concat ", " (List.map string_of_int bs)) ^ "]"
```
Comment (Nadrieril) [inline charon-ml/src/PrintValues.ml:37]: There seems to be the same function defined twice; can we reuse one of them?

File: charon-ml/src/NameMatcher.ml [hunk 3]
```diff
@@ -63,6 +63,8 @@ let literal_to_string (c : print_config) (l : literal) : string =
           raise (Failure "TODO")
       | TkPretty -> "'" ^ String.make 1 x ^ "'"
       | TkName -> String.make 1 x)
+  | LStr s -> "\"" ^ s ^ "\""
+  | LByteStr bs -> "[" ^ String.concat ", " (List.map string_of_int bs) ^ "]"
```
Comment (Nadrieril) [inline charon-ml/src/NameMatcher.ml:67]: There seems to be the same function defined twice; can we reuse one of them?

**Comment by Nadrieril** [comment]
The problem with this is that aeneas does not have a pretty error path where it uses `Literal`, so to update aeneas I have to either ilplement it correctly for all backends or panic.

File: charon-ml/src/PrintValues.ml [hunk 4]
```diff
@@ -33,3 +33,5 @@ let literal_to_string (lit : literal) : string =
   | VScalar sv -> scalar_value_to_string sv
   | VBool b -> Bool.to_string b
   | VChar c -> String.make 1 c
+  | VStr s -> "\"" ^ s ^ "\""
+  | VByteStr bs -> "[" ^ String.concat ", " (List.map string_of_int bs) ^ "]"
```
[the remark this answers is not in the document]
Comment (sonmarcho) [reply charon-ml/src/PrintValues.ml:37]: This is a "regular" pretty printing function used to pretty print the (U)LLBC ast. This looks good to me.

**Review by sonmarcho — CHANGES_REQUESTED** [review]

File: charon-ml/name_matcher_parser/Ast.ml [hunk 5]
```diff
@@ -22,7 +22,14 @@ class ['self] map_pattern_base =
...
 
 type var = VarName of string | VarIndex of int
-and literal = LInt of big_int | LBool of bool | LChar of char
+
+and literal =
+  | LInt of big_int
+  | LBool of bool
+  | LChar of char
+  | LStr of string
+  | LByteStr of int list
```
[the remark this answers is not in the document]
Comment (Nadrieril) [reply charon-ml/name_matcher_parser/Ast.ml:31]: What's a "pattern"?

File: charon/tests/ui/unsupported/plain-panic-str.out [hunk 6]
```diff
@@ -0,0 +1,15 @@
+thread 'rustc' panicked at 'Expected a next block after the call to core::panicking::panic_fmt.
```
[the remark this answers is not in the document]
Comment (Nadrieril) [reply charon/tests/ui/unsupported/plain-panic-str.out:1]: So we ignore the argument to `panic!()`?

**Review by sonmarcho — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**